src/controllers/authController.js

```diff
@@ -0,0 +1,156 @@
+'use strict';
+
+const bcrypt = require('bcrypt');
+const userSevices = require('../services/userServices.js');
+const jwtServices = require('../services/jwtServices.js');
+const { User } = require('../models/user.js');
+const { ApiError } = require('../exeption/apiError.js');
+const tokenSevices = require('../services/tokenServices.js');
+const validate = require('../utils/validate.js');
+
+const register = async(req, res) => {
+  const { name, email, password } = req.body;
+
+  const errors = {
+    email: validate.email(email),
+    password: validate.password(password),
+    name: validate.username(name),
+  };
+
+  if (!errors.email || !errors.password || !errors.name) {
+    throw ApiError.badRequest('Bad requst', errors);
+  }
+
+  await userSevices.register(name, email, password);
```

Comment:
Consider using the try-catch block here. In some cases, we should send the 500 error if we have issues with the database or something else during the process.

Reply:
Consider implementing this logic further in the code